### PR TITLE
Renaming changes 

### PR DESCRIPTION
**Pull request by sima-zhu**
1. Boolean fields
2. Varargs
3. RequestOptions

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/RequestOptions.java [hunk 1]
```diff
@@ -16,27 +16,27 @@
      * The tracking ID sent with the request to help with debugging.
      */
     @JsonProperty(value = "")
-    private UUID xMsClientRequestId;
+    private UUID clientRequestId;
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/RequestOptions.java:19]: Is the value set here actually setting `client-request-id` or `x-ms-client-request-id`?

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/LimitTokenFilterConverter.java [hunk 2]
```diff
@@ -46,7 +46,7 @@ public static LimitTokenFilter map(com.azure.search.documents.indexes.implementa
         Integer maxTokenCount = obj.getMaxTokenCount();
         limitTokenFilter.setMaxTokenCount(maxTokenCount);
 
-        Boolean consumeAllTokens = obj.isConsumeAllTokens();
+        Boolean consumeAllTokens = obj.isAllTokensConsumed();
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/LimitTokenFilterConverter.java:49]: In the plural cases should we use `are`? `areAllTokensConsumed`.

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/EntityRecognitionSkillConverter.java [hunk 3]
```diff
@@ -100,7 +100,7 @@ public static EntityRecognitionSkill map(com.azure.search.documents.indexes.impl
         String description = obj.getDescription();
         entityRecognitionSkill.setDescription(description);
 
-        Boolean includeTypelessEntities = obj.isIncludeTypelessEntities();
+        Boolean includeTypelessEntities = obj.isTypelessEntitiesIncluded();
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/EntityRecognitionSkillConverter.java:103]: `areTypelessEnititiesIncluded`?

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/PhoneticTokenFilterConverter.java [hunk 4]
```diff
@@ -24,7 +24,7 @@ public static PhoneticTokenFilter map(com.azure.search.documents.indexes.impleme
         String name = obj.getName();
         phoneticTokenFilter.setName(name);
 
-        Boolean replaceOriginalTokens = obj.isReplaceOriginalTokens();
+        Boolean replaceOriginalTokens = obj.isOriginalTokensReplaced();
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/PhoneticTokenFilterConverter.java:27]: `areOriginalTokensReplaced`?

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/StopwordsTokenFilterConverter.java [hunk 5]
```diff
@@ -59,10 +59,10 @@ public static StopwordsTokenFilter map(com.azure.search.documents.indexes.implem
         String name = obj.getName();
         stopwordsTokenFilter.setName(name);
 
-        Boolean removeTrailingStopWords = obj.isRemoveTrailingStopWords();
+        Boolean removeTrailingStopWords = obj.isTrailingStopWordsRemoved();
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/StopwordsTokenFilterConverter.java:62]: `areTrailingStopwordsRemoved`?

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/WordDelimiterTokenFilterConverter.java [hunk 6]
```diff
@@ -74,7 +74,7 @@ public static WordDelimiterTokenFilter map(com.azure.search.documents.indexes.im
         String name = obj.getName();
         wordDelimiterTokenFilter.setName(name);
 
-        Boolean catenateNumbers = obj.isCatenateNumbers();
+        Boolean catenateNumbers = obj.isNumbersCatenated();
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/WordDelimiterTokenFilterConverter.java:77]: `areNumbersCatenated`?

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/WordDelimiterTokenFilterConverter.java [hunk 7]
```diff
@@ -100,10 +100,10 @@ public static WordDelimiterTokenFilter map(com.azure.search.documents.indexes.im
         Boolean preserveOriginal = obj.isPreserveOriginal();
         wordDelimiterTokenFilter.setPreserveOriginal(preserveOriginal);
 
-        Boolean catenateAll = obj.isCatenateAll();
+        Boolean catenateAll = obj.catenateAll();
         wordDelimiterTokenFilter.setCatenateAll(catenateAll);
 
-        Boolean catenateWords = obj.isCatenateWords();
+        Boolean catenateWords = obj.isWordsCatenated();
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/WordDelimiterTokenFilterConverter.java:106]: `areWordsCatenated`?

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/implementation/models/PhoneticTokenFilter.java [hunk 8]
```diff
@@ -33,7 +33,7 @@
      * tokens. If false, encoded tokens are added as synonyms. Default is true.
      */
     @JsonProperty(value = "replace")
-    private Boolean replaceOriginalTokens;
+    private Boolean originalTokensReplaced;
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/implementation/models/PhoneticTokenFilter.java:36]: Is this a generated class? It's in implementation models but I know many of these classes were copied from generated code.

Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/implementation/models/PhoneticTokenFilter.java:36]: Good catch. Reverts all changes under impl

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/RequestOptions.java (hunk 1 above)
Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/RequestOptions.java:19]: RequestOptions will match to internal RequestOptions, the name over there does not change. The JsonProperty here does not take affect.

Comment (sima-zhu) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/RequestOptions.java:19]: We just change the API exposed to customer

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/SynonymTokenFilterConverter.java [hunk 9]
```diff
@@ -53,15 +53,15 @@ public static SynonymTokenFilter map(com.azure.search.documents.indexes.implemen
         String name = obj.getName();
         synonymTokenFilter.setName(name);
 
-        Boolean expand = obj.isExpand();
+        Boolean expand = obj.getExpand();
```
Comment (alzimmermsft) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/converters/SynonymTokenFilterConverter.java:56]: Should this be `isExpand`?

**Pull request merged**